pages/studentAssessmentExam/studentAssessmentExam.js

```diff
@@ -11,22 +11,93 @@ var sql = sqlLoader.loadSqlEquiv(__filename);
 
 router.get('/', function (req, res, next) {
   if (res.locals.assessment.type !== 'Exam') return next();
+  var params = {
+    assessment_id: res.locals.assessment.id,
+    user_id: res.locals.user.user_id,
+  };
   if (res.locals.assessment.multiple_instance) {
-    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+    if (res.locals.assessment.group_work) {
```

Comment:
This same chunk of code is repeated three times in this file, and appears to be copy/pasted from `studentAssessmentHomework.js`. Let's refactor this into a shared function.